grpc-circuitbreaker-utils/src/main/java/org/hypertrace/circuitbreaker/grpcutils/CircuitBreakerConfigParser.java

```diff
@@ -38,20 +38,19 @@ public static <T> CircuitBreakerConfiguration.CircuitBreakerConfigurationBuilder
       builder.enabled(config.getBoolean(ENABLED));
     }
 
-    if (config.hasPath(DEFAULT_THRESHOLDS)) {
-      builder.defaultThresholds(
-          buildCircuitBreakerThresholds(config.getConfig(DEFAULT_THRESHOLDS)));
-    } else {
-      builder.defaultThresholds(buildCircuitBreakerDefaultThresholds());
-    }
-
     Map<String, CircuitBreakerThresholds> circuitBreakerThresholdsMap =
         config.root().keySet().stream()
             .filter(key -> !NON_THRESHOLD_KEYS.contains(key)) // Filter out non-threshold keys
             .collect(
                 Collectors.toMap(
                     key -> key, // Circuit breaker key
                     key -> buildCircuitBreakerThresholds(config.getConfig(key))));
+
+    if (!config.hasPath(DEFAULT_THRESHOLDS)) {
+      builder.defaultThresholds(buildCircuitBreakerDefaultThresholds());
+      circuitBreakerThresholdsMap.put(DEFAULT_THRESHOLDS, buildCircuitBreakerDefaultThresholds());
```

Comment:
Why are we writing this into the map? The the default thresholds don't need to be accessed by key - that should be what `builder.defaultThresholds` is for. Also note if `DEFAULT_THRESHOLDS` path _does_ exist it would make it into the map, but it would not be set to `builder.defaultThresholds`. I would expect this logic to look more like
```java
builder.defaultThresholds(
  config.hasPath(DEFAULT_THRESHOLDS) ?
    buildCircuitBreakerThresholds(config.getConfig(DEFAULT_THRESHOLDS)) : 
    buildCircuitBreakerDefaultThresholds());
```

The config key then is not a concern of the specific impl. If you really want to use a magic key, then `defaultThresholds` would be removed, we wouldn't want both since it's two ways of conveying the same thing - one just decouples concerns.

Reply:
done

Reply:
This is done mainly to handle where circuit breaker is disabled by default but enabled for only tenant, in CircuitBreakerProvider we check enabled or based on CircuitConfigMap only, so thats why adding to map.